third_party/amd/lib/TritonAMDGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMFMA.cpp

```diff
@@ -260,7 +260,7 @@ Value convertLayout(int opIdx, ConversionPatternRewriter &rewriter,
 
   Value spatialWarpId = AMD::getWarpIdInBlock(
       rewriter, loc, linearWarpId, warpsPerCTA, mfmaInstrNonK,
-      shape[nonKDimIdx], nonKDimIdx, mfmaLayout.getDefaultOrder());
+      shape[nonKDimIdx], nonKDimIdx, getWarpOrder(mfmaLayout, shape));
```

Comment:
@lezcano The test failure is due to incorrect warpOrder. We need to have [1, 0] as the warpOrder for both operands. I forced the warpOrder to be [1, 0] in https://github.com/triton-lang/triton/pull/6188 and all tests can pass. However, https://github.com/triton-lang/triton/pull/6188 is not a proper fix. I think here we want the "shape" of the result tensor, which is the tensor this `mfmaLayout` is attached to, rather than the shape of the operand tensor.
The challenge is that the dst of `local_load` only contains `mfmaLayout` but not the shape information of the dot result. Maybe we need to start from the `local_load` and find the corresponding `dot` and get the shape. Let me know if you have something else in mind.

Reply:
@Jokeren  We can merge this one for now since we plan to get rid of these whole `SharedToDotOperandXXX.cpp` path in the near future after I figured the regression issues.